topic.go

```diff
@@ -148,13 +174,31 @@ func WithReadiness(ready RouterReady) PubOpt {
 // Close closes down the topic. Will return an error unless there are no active event handlers or subscriptions.
 // Does not error if the topic is already closed.
 func (t *Topic) Close() error {
+	t.mux.Lock()
+	defer t.mux.Unlock()
+	if t.closed {
+		return nil
+	}
+
 	req := &rmTopicReq{t, make(chan error, 1)}
 	t.p.rmTopic <- req
```

Comment:
https://github.com/libp2p/go-libp2p-pubsub/issues/226

Reply:
This was true for all previous PubSub functions and the new Topic ones. I've pushed an update, lmk if that's what you had in mind.

Three questions about fixing this:
1) Do I want custom `ErrPubSubClosed` messages or is the context error enough (both for the PubSub functions and for the Topic functions)?
2) Should I set `Topic.closed=true` on any/all of the pubsub context errors?
3) It seems more reasonable to return an empty array then nil for `GetTopics` and `ListPeers` right?

Reply:
@vyzo  any thoughts on the first 2 questions here? Would like to get this pushed out ASAP since the `Close` bug is particularly bad.

Reply:
1. either works, it's a very small thing.
2. probably not, I don't see what we would gain from this.